internal/exec/stages/disks/partition_s390_disk.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +build s390x
+
+// s390x variant of partitionDisk, checks cmdline for rd.dasd and
+// if set calls the DASD partitioning function else fall back to
+// partitionDevDisk
+
+package disks
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/coreos/ignition/v2/config/v3_2_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
+)
+
+// hasDasdDisks checks the cmdline to see if `rd.dasd=` has been set. If it has,
+// it means that we are dealing with DASD devices. If not, fall back to using the
+// regular way of partitioning using sgdisk
+func hasDasdDisks() (bool, error) {
```

Comment:
the intent of this was to identify whether the system has DASD disks so we can use the DASD commands to format and partition.I didn't think of checking for the disk to be present because anyways the wait for disk will fail in that case. There was no good equivalent to say - "this zVM uses DASD disks". Also if this is bogus, the coreos-installer itself would bomb out so this is a pretty safe check.

Reply:
hmm..that's a good point about systems with mixed disks..not sure if that is a valid configuration for zVMs..but then for the live case we need another method anyway. I could check for the device - if the name is `/dev/dasdx`, then it's a dasd. the other case is if the user specifies by-path - for dasds it would be `/dev/disk/by-path/ccw-0.0.0121` and for SCSI disks it is `/dev/disk/by-path/ccw-0.0.0120-zfcp` so i guess we could differentiate this way.

Reply:
For the `by-path` case, wouldn't it be cleaner to resolve the symlink and check for `/dev/dasd*`?  That way, any other `/dev/disk/by-*` links will also work.

Reply:
changed this to follow the alias name to the real device name , i.e /dev/dasdx..